ocaml/xapi/xapi.ml

```diff
@@ -444,7 +447,60 @@ let attempt_pool_hello my_ip =
               [localhost_uuid] ;
             Some Permanent
         | `ok ->
-            None
+            let obj_uuid = Helpers.get_localhost_uuid () in
+            let message_name, _ =
+              Api_messages
+              .xapi_startup_blocked_as_version_higher_than_coordinator
+            in
+            let existing_alerts =
+              Client.Client.Message.get_all_records ~rpc ~session_id
+              |> List.filter (fun (_ref, record) ->
+                     record.API.message_obj_uuid = obj_uuid
+                     && record.API.message_name = message_name
+                 )
+            in
+            let compare_xapi_version_with_coordinator ~__context =
+              let coordinator_ref = Helpers.get_master ~__context in
+              let coordinator_xapi_ver =
+                Client.Client.Host.get_software_version ~rpc ~session_id
+                  ~self:coordinator_ref
+                |> List.assoc "xapi_build"
+              in
+              Xapi_version.compare_version Xapi_version.version
+                coordinator_xapi_ver
+            in
+            if compare_xapi_version_with_coordinator ~__context > 0 then (
+              ( if existing_alerts = [] then
+                  let hostname =
+                    Db.Host.get_hostname ~__context
```

Comment:
I think you want `get_name_label`.

Reply:
OK, changed.
Would you please share what's difference between the two fields of a host: `hostname` and `name_label`? It is strange that the comment for `name_label` field is:
`param_doc= "The name of the new storage repository"`
And from current code, I see both `hostname` and `name_label` are used in logging.

Reply:
`Host.name_label` is a read/write field for users to set to any name they like for their hosts. The `Host.hostname` is the network hostname.